[WIP] ROSA topic map restructure (#41923)
openshift/openshift-docs: rosa_cluster_admin/rosa_nodes/rosa-nodes-machinepools-about.adoc

:_content-type: ASSEMBLY
include::_attributes/attributes-openshift-dedicated.adoc[]
[id="rosa-nodes-machinepools-about"]

> = About machine pools

Reviewer: Double-check that you're happy with the title's scope.

:context: rosa-nodes-machinepools-about

toc::[]

{product-title} uses machine pools as an elastic, dynamic provisioning method on top of your cloud infrastructure.

The primary resources are machines, machine sets, and machine pools.

> == Machines

Reviewer: This sounds like a concept module.

And the headings in this topic should have anchors.

A machine is a fundamental unit that describes the host for a worker node.

== Machine sets

> `MachineSet` resources are groups of machines. If you need more machines or must scale them down, this is done by configuring the number of replicas in the machine pool to which the machine sets belong.

Reviewer: ```suggestion
`MachineSet` resources are groups of machines. If you need more machines or must scale them down, change the number of replicas in the machine pool to which the machine sets belong.
```

ifdef::openshift-rosa[]
Machine sets are not directly modifiable in ROSA.
endif::[]

== Machine pools
Machine pools are a higher level construct to machine sets.

> A machine pool creates machine sets that are all clones of the same configuration across availability zones. Machine pools perform all of the host node provisioning management actions on a worker node. If you need more machines or must scale them down, change the number of replicas in the machine pool to meet your compute needs. Scaling can be configured manually or you can set autoscaling.

Reviewer: ```suggestion
A machine pool creates machine sets that are all clones of the same configuration across availability zones. Machine pools perform all of the host node provisioning management actions on a worker node. If you need more machines or must scale them down, change the number of replicas in the machine pool to meet your compute needs. You can manually configure scaling or set autoscaling.
```

> By default, a cluster is created with one machine pool. Additional machine pools can be added later to an existing cluster and you can modify the default machine pool. Machine pools can also be deleted.

Reviewer: ```suggestion
By default, a cluster is created with one machine pool. You can add additional machine pools to an existing cluster, modify the default machine pool, and delete machine pools.
```

Multiple machine pools can exist on a single cluster, and they can each have different types or different size nodes.

== Machine pools in multiple zone clusters

> When you create a machine pool in a multiple availability zone (Multi-AZ) cluster, that one machine pool has 3 zones. The machine pool, in turn, creates a total of 3 machine sets - one machine set for each zone in the cluster. Each of those machine sets manages one or more machines in its respective availability zone.

Reviewer: Please be consistent with spelling out small numbers or using numerals.

> If you create a new Multi-AZ cluster, the machine pools are replicated to those zones automatically. If you add a machine pool to an existing Multi-AZ, the new pool is automatically created in those zones. Similarly, deleting a machine pool will delete it from all zones.

Reviewer: ```suggestion
If you create a new Multi-AZ cluster, the machine pools are replicated to those zones automatically. If you add a machine pool to an existing Multi-AZ <noun_goes_here>, the new pool is automatically created in those zones. Similarly, deleting a machine pool will delete it from all zones.
```
